saleor/static/dashboard-next/siteSettings/components/SiteSettingsKeyDialog/SiteSettingsKeyDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import TextField from "@material-ui/core/TextField";
+import * as React from "react";
+
+import Form, { FormProps } from "../../../components/Form";
+import { FormSpacer } from "../../../components/FormSpacer";
+import SingleSelectField from "../../../components/SingleSelectField";
+import i18n from "../../../i18n";
+import { translatedAuthorizationKeyTypes } from "../../../misc";
+import { AuthorizationKeyType } from "../../../types/globalTypes";
+
+export interface SiteSettingsKeyDialogForm {
+  key: string;
+  password: string;
+  type: AuthorizationKeyType;
+}
+
+export interface SiteSettingsKeyDialogProps
+  extends Pick<
+      FormProps<SiteSettingsKeyDialogForm>,
+      Exclude<keyof FormProps<SiteSettingsKeyDialogForm>, "children">
+    > {
+  open: boolean;
+  onClose: () => void;
+}
+
+const SiteSettingsKeyDialog: React.StatelessComponent<
+  SiteSettingsKeyDialogProps
+> = ({ errors, initial, open, onClose, onSubmit }) => (
+  <Dialog maxWidth="xs" open={open}>
+    <Form initial={initial} onSubmit={onSubmit} errors={errors}>
+      {({ change, data, errors }) => (
+        <>
+          <DialogTitle>
+            {i18n.t("Add New Authorization Key", {
+              context: "modal title"
+            })}
+          </DialogTitle>
+          <DialogContent>
+            <SingleSelectField
+              choices={Object.keys(translatedAuthorizationKeyTypes()).map(
+                key => ({
+                  label: translatedAuthorizationKeyTypes()[key],
```

Comment:
This will evaluate `translatedAuthorizationKeyTypes` in each iteration of the `map` loop.